io.openems.edge.io.shelly/src/io/openems/edge/io/shelly/shelly25/IoShelly25Impl.java

```diff
@@ -75,20 +89,19 @@ public BooleanWriteChannel[] digitalOutputChannels() {
 	public String debugLog() {
 		var b = new StringBuilder();
 		var i = 1;
-		for (WriteChannel<Boolean> channel : this.digitalOutputChannels) {
+		for (BooleanWriteChannel channel : this.digitalOutputChannels) {
 			String valueText;
 			var valueOpt = channel.value().asOptional();
 			if (valueOpt.isPresent()) {
-				valueText = valueOpt.get() ? "x" : "-";
+				valueText = valueOpt.get() ? "ON" : "OFF";
 			} else {
-				valueText = "?";
+				valueText = "Unknown";
```

Comment:
Because we have it on the other Shelly Devices aswell personally i would also not make "ON" and "OFF" but x or -as we have it on KM-Tronic Relays but in this case:

Changed because of keeping up with the other Shelly Devices (Shelly 3EM and Sehlly Plug for example) See here:

https://github.com/OpenEMS/openems/blob/f885ae9368134142d7d899cb2e3461eb58ada4e8/io.openems.edge.io.shelly/src/io/openems/edge/io/shelly/shellyplug/IoShellyPlugImpl.java#L107C1-L108C1

and:

https://github.com/OpenEMS/openems/blob/f885ae9368134142d7d899cb2e3461eb58ada4e8/io.openems.edge.io.shelly/src/io/openems/edge/io/shelly/shelly3em/IoShelly3EmImpl.java#L118